### PR TITLE
Change price field in TaskToCompute

### PR DESCRIPTION
**Pull request by Krigpl**
Use value from `WantToComputeTask` and change its meaning to total amount to be paid, instead of rate per hour.

### CODE REVIEW

**Comment by shadeofblue** [comment]
@Krigpl did we decide anything with regards to this pull request? is there a complementary pull request for golem core?

**Comment by Krigpl** [comment]
@shadeofblue  It's postponed until `TaskHeader` is included in `WantToComputeTask`

File: golem_messages/message/tasks.py [hunk 1]
```diff
@@ -292,12 +291,16 @@ def provider_ethereum_public_key(self):
...
+    @property
+    def price(self):
+        return self.want_to_compute_task.price
+
     @base.verify_slot('want_to_compute_task', WantToComputeTask)
     def deserialize_slot(self, key, value):
         value = super().deserialize_slot(key, value)
         if key == 'compute_task_def':
             value = ComputeTaskDef(value)
-        if key in ('price', 'size'):
```
Comment (jiivan) [inline golem_messages/message/tasks.py]: Please move price validation to WTC along with price slot.

**Comment by Krigpl** [comment]
I'd somewhat prefer a different name for this property though - since it doesn't represent rate per hour anymore.

**Comment by jiivan** [comment]
> I'd somewhat prefer a different name for this property though - since it doesn't represent rate per hour anymore.

IMO WTC.price is more misleading. I propose `.rate_per_hour`

**Comment by Krigpl** [comment]
> IMO WTC.price is more misleading. I propose .rate_per_hour

Following this convention TaskHeader.price should be changed as well. Most of Golem assumes hourly rate when using `price`

**Comment by jiivan** [comment]
> Following this convention TaskHeader.price should be changed as well. Most of Golem assumes hourly rate when using `price`

It should.

**Comment by shadeofblue** [comment]
actually, I think this pull request no longer makes sense

**Comment by Krigpl** [comment]
@shadeofblue Why do you think that?

**Pull request closed without merging**